test/printf.jl

```diff
@@ -3,46 +3,49 @@
 # printf
 # int
 @test (@sprintf "%d" typemax(Int64)) == "9223372036854775807"
-@test (@sprintf "%i" 42) == "42"
-@test (@sprintf "%u" 42) == "42"
-@test (@sprintf "Test: %i" 42) == "Test: 42"
-@test (@sprintf "%#x" 42) == "0x2a"
-@test (@sprintf "%#o" 42) == "052"
-@test (@sprintf "%X" 42) == "2A"
-@test (@sprintf "%X" 42) == "2A"
-@test (@sprintf "% i" 42) == " 42"
-@test (@sprintf "%+i" 42) == "+42"
-@test (@sprintf "%4i" 42) == "  42"
-@test (@sprintf "%-4i" 42) == "42  "
-# float
-@test (@sprintf "%7.2f" 1.2345) == "   1.23"
-@test (@sprintf "%-7.2f" 1.2345) == "1.23   "
-@test (@sprintf "%07.2f" 1.2345) == "0001.23"
-@test (@sprintf "%.0f" 1.2345) == "1"
-@test (@sprintf "%#.0f" 1.2345) == "1."
+for (fmt, val) in (("%i", "42"), ("%u", "42"), ("Test: %i", "Test: 42"),
+                   ("%#x", "0x2a"), ("%#o", "052"), ("%X", "2A"), ("%X", "2A"),
+                   ("% i", " 42"), ("%+i", "+42"), ("%4i", "  42"), ("%-4i", "42  "))
+    @test( @eval(@sprintf($fmt, 42) == $val))
+end
+
+# float / BigFloat
+for (fmt, val) in (("%7.2f", "   1.23"), ("%-7.2f", "1.23   "), ("%07.2f", "0001.23"), ("%.0f", "1"), ("%#.0f", "1."), ("%.4e", "1.2345e+00")),
+	num in (1.2345, big"1.2345")
+    @test( @eval(@sprintf($fmt, $num) == $val))
+end
 # Inf / NaN handling
 @test (@sprintf "%f" Inf) == "Inf"
 @test (@sprintf "%f" NaN) == "NaN"
+@test (@sprintf "%f" big"Inf") == "Inf"
+@test (@sprintf "%f" big"NaN") == "NaN"
 # scientific notation
-@test (@sprintf "%.4e" 1.2345) == "1.2345e+00"
 @test (@sprintf "%.0e" 3e142) == "3e+142"
 @test (@sprintf "%#.0e" 3e142) == "3.e+142"
+@test (@sprintf "%.0e" big"3e142") == "3e+142"
+@test (@sprintf "%#.0e" big"3e142") == "3.e+142"
 # %g
-@test( @sprintf( "%.6g", 12345678. ) == "1.23457e+07" )
-@test( @sprintf( "%.6g", 1234567.8 ) == "1.23457e+06" )
-@test( @sprintf( "%.6g", 123456.78 ) == "123457" )
-@test( @sprintf( "%.6g", 12345.678 ) == "12345.7" )
-@test( @sprintf( "%10.5g", 123.4 ) == "     123.4")
-@test( @sprintf( "%+10.5g", 123.4 ) == "    +123.4")
-@test( @sprintf( "% 10.5g", 123.4 ) == "     123.4")
-@test( @sprintf( "%#10.5g", 123.4 ) == "    123.40")
-@test( @sprintf( "%-10.5g", 123.4 ) == "123.4     ")
-@test( @sprintf( "%-+10.5g", 123.4 ) == "+123.4    ")
+for (val, res) in ((12345678., "1.23457e+07"), (1234567.8, "1.23457e+06"), (123456.78, "123457"), (12345.678, "12345.7"), (12340000.0, "1.234e+07"))
```

Comment:
There were zero line breaks in the original file, I can add them before the sections (where there are `#` comments) if you think it would make it easier to read.